src/core/TrainPart.cpp

```diff
@@ -80,6 +79,9 @@ namespace godot {
                 }
                 train_controller_node = nullptr;
             } break;
+            default:
+                DEBUG("TrainController::_ready() This notification type is not implemented");
```

Comment:
There should be no warning message, because it will generate spam messages - not all notifications must be handled.